src/cli/repl/server/messages/lineage.ts

```diff
@@ -0,0 +1,28 @@
+import type { SingleSlicingCriterion } from '../../../../slicing/criterion/parse'
+import type { IdMessageBase, MessageDefinition } from './messages'
+import type { NodeId } from '../../../../r-bridge/lang-4.x/ast/model/processing/node-id'
+import Joi from 'joi'
+
+export interface LineageRequestMessage extends IdMessageBase {
+	type:      'request-lineage',
+	/** The {@link FileAnalysisRequestMessage#filetoken} of the file/data */
+	filetoken: string,
+	/** The criterion to start the lineage from */
+	criterion: SingleSlicingCriterion,
+}
+
+export const requestLineageMessage: MessageDefinition<LineageRequestMessage> = {
+	type:   'request-lineage',
+	schema: Joi.object({
+		type:      Joi.string().valid('request-lineage').required(),
+		id:        Joi.string().optional(),
+		filetoken: Joi.string().required(),
+		criterion: Joi.string().required()
+	})
+}
+
+export interface LineageResponseMessage extends IdMessageBase {
+	type:    'response-lineage',
+	/** The lineage of the given criterion */
```

Comment:
maybe add information about the order? 
Is the 'root' element at the start or the end of the list?